File: algorithms/0260. Single Number III.cpp
class Solution
{
public:
    vector<int> singleNumber(vector<int>& nums)
    {
        long xr = 0;
        
        for (int num : nums)
        {
            xr ^= num;
        }
        
        int mask = xr & (-xr);
        
        int a = 0;
        int b = 0;
        
        for (int num : nums)
        {
            if (num & mask)
            {
                a ^= num;
            }
            else
            {
                b ^= num;
            }
        }
        
        return { a, b };
    }
};

File: 0260. Single Number III.cs
public class Solution
{
    public int[] SingleNumber(int[] nums)
    {
        var xor = 0;

        foreach (var num in nums)
        {
            xor ^= num;
        }

        var a = 0;
        var b = 0;

        var mask = xor & (-xor);

        foreach (var num in nums)
        {
            if ((num & mask) == 0)
            {
                a ^= num;
            }
            else
            {
                b ^= num;
            }
        }

        return [a, b];
    }
}